statefun-tasks/src/main/java/com/sbbsystems/statefun/tasks/PipelineFunction.java

```diff
@@ -39,12 +49,27 @@ public class PipelineFunction implements StatefulFunction {
     @Override
     public void invoke(Context context, Object input) {
         try (var ignored = TimedBlock.of(LOG::info, "Invoking function {0}", context.self())) {
+            if (MessageTypes.isType(input, TaskRequest.class)) {
```

Comment:
A bit of duplicated code.  Could replace with a type implementing AutoClose similar to TimedBlock?

try (Signal ignored = Signal.for(input, context.self().id(), this.callbackFunctionType)) {
   ... process message handlers 
}

On instantiation send the PIPELINE_STARTING signal and on auto close send the BATCH_PROCESSED signal.

Reply:
At the moment, the BATCH_PROCESSED signal is only sent when this function processed a ResultsBatch. Maybe it should be the responsibility of the handler for ResultsBatch messages.

I think I should move this logic into handlers anyway.